perfkitbenchmarker/pkb.py

```diff
@@ -195,6 +195,51 @@
 events.initialization_complete.connect(traces.RegisterAll)
 
 
+def _CreateBenchmarkRunList():
+  """Create a list of configs and states for each benchmark run to be scheduled.
+
+  Returns:
+    list of (args, run_status_list) pairs. Contains one pair per benchmark run
```

Comment:
Passing around tuples of arguments and lists of run statuses is getting a bit hard to keep track of. It might make it a bit more readable if we bound the arguments directly to the RunBenchmark function and just passed around a bunch of partials that we called later. Making the benchmark status an attribute of the BenchmarkSpec (or if that's not appropriate, defining a Benchmark class that can keep track of status and stage etc) would probably also help. This PR doesn't make things any worse than they were, so I'm fine skipping this for now, but it would be nice if this could eventually get cleaned up.